samples/transactional-session/aspnetcore-webapi/Core_8/WebApplication/Program.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Data.SqlClient;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using NServiceBus;
+using NServiceBus.Persistence;
+
+public class Program
+{
+    const string ConnectionString = @"Server=(localdb)\MSSQLLocalDB;Integrated Security=True;";
+
+    public static void Main()
+    {
+        using (var myDataContext = new MyDataContext(new DbContextOptionsBuilder<MyDataContext>()
+                   .UseSqlServer(new SqlConnection(ConnectionString))
+                   .Options))
+        {
+            myDataContext.Database.EnsureCreated();
+        }
+
+        var host = Host.CreateDefaultBuilder()
+
+            #region txsession-nsb-configuration
+            .UseNServiceBus(context =>
+            {
+                var endpointConfiguration = new EndpointConfiguration("Samples.ASPNETCore.Sender");
+                var transport = endpointConfiguration.UseTransport(new LearningTransport { TransportTransactionMode = TransportTransactionMode.ReceiveOnly });
+                endpointConfiguration.EnableInstallers();
+
+                var persistence = endpointConfiguration.UsePersistence<SqlPersistence>();
+                persistence.SqlDialect<SqlDialect.MsSqlServer>();
+                persistence.ConnectionBuilder(() => new SqlConnection(ConnectionString));
+
+                endpointConfiguration.EnableOutbox();
+                endpointConfiguration.EnableTransactionalSession();
```

Comment:
Could we somehow inlay `EnableOutbox` into the other API so that you only have to write one?

Reply:
Given that it also works with batching this would be not such a good idea